Add lshw partial implementation

avinal opened the pull request:
This merge request does the following:-
- Adds  partial `lshw` command implementation
- Closes #1717

anton-bondarev commented:
@avinal Thanks. Could you also please move `cpuinfo` and `lsblk` parts to separate procedures. And format a little more carefully.

avinal commented:
Hello @anton-bondarev Sure, actually I had it added in my development branch, somehow missed in the pull request. I will add it. 

If you insist a more cleaner code I can refactor this code into totally seperate functions for each of the command and then we can just call them in `main`. 

Also I tried keeping the formatting as close to the embox formatting, don't know where I missed. I am using `llvm` format. I will do that again and let you know if I need some help. 

Thanks :)

anton-bondarev commented on src/cmds/hardware/lshw/lshw.c:
```diff
@@ -0,0 +1,441 @@
+}
+
+static void show_usb_desc_configuration(struct usb_dev *usb_dev) {
+	struct usb_desc_configuration *config = \
```
Why do you use '\' here?

avinal replied:
I didn't get your point. This is the exact code, I didn't change anything. But the backslash can be removed.

anton-bondarev replied:
Oh, sorry. I didn't understand that it is a cloned code. But in any way, if you use code, it is yours

anton-bondarev commented:
>  I will add it.

Ok.

> If you insist a more cleaner code I can refactor this code into totally seperate functions for each of the command and then we can just call them in `main`.

Yes, I think It will look better and easier to read, Please do it if you could

> Also I tried keeping the formatting as close to the embox formatting, don't know where I missed. I am using `llvm` format. I will do that again and let you know if I need some help.

Thanks. Your last commit is that I meant

avinal commented:
> Yes, I think It will look better and easier to read, Please do it if you could

On it, I got an unexpected error too I am fixing that too.

avinal commented:
Hello @anton-bondarev 
Sorry for the delay, I have cleaned the code and refactored into seperate functions for each command, and removed unnecessary flag options since lshw is generally run in verbose mode. 

I have also tried to keep the formatting strictly as described in Wiki. Hope this is complete now :)

anton-bondarev reviewed (approved):
LGTM

anton-bondarev commented:
Hello @avinal Thanks. It's OK for now. I've approved it. Please rebase and squash commits if it's necessary

avinal commented:
I have rebased and squashed the commits :)

anton-bondarev commented:
> I have rebased and squashed the commits :)

Thanks. Could you please split on two commits, first lshw command and second inserting it to mods.conf?

avinal commented:
Done, now I think the version control is really difficult to master than coding itself 😂.

anton-bondarev commented:
@avinal Thanks.
git workflow is the simplest thing. Embox is a operating system :)
